### PR TITLE
directly input and output to /dev/tty (or CON:) while doing --debug stuff

### PR DESCRIPTION
**Pull request by shulinye**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 *.pyc
-stocks.csv
+test.py
+test.ol.py
```
Comment (andersk) [inline .gitignore]: This looks like a mistake.

**Comment by andersk** [comment]
It’s really unusual for programs to need to access `/dev/tty` directly. Why isn’t stderr good enough?

**Comment by shulinye** [comment]
Several things.

1. Someone who just wants to muck around with the program is likely to make
a test.py and then a test.ol.py .  At least, test.py is my naming
convention for "I want to see how this module works and need to write a
little script
2. I'm already reading from /dev/tty , figured I might as well write to it.
Also, uh, I'm really not a fan of putting not-errors on sys.stderr.
Although now that I'm thinking of it, those exceptions should really go to
stderr.

**Comment by andersk** [comment]
1. If you want `test.py` and `test.ol.py` ignored because you are using them, did you know that you can put local `gitignore` lines in `.git/info/exclude` (per-repository) or `~/.config/git/ignore` (global) without needing to commit them to the repository? See `git help gitignore`. Also, `stocks.csv` is in `.gitignore` because it’s written by `tests/16lines.py`.

2. I’m guessing your proposed use case is something like
`echo 'x = raw_input("Name: "); print "Hello,", x' | python main.py --debug`
where you want the input to come from the user, not the `echo` command? But you can already do this:
`python main.py --debug <(echo 'x = raw_input("Name: "); print "Hello,", x') /dev/null`
because `<()` comes in on its own file descriptor. On the other hand, by forcefully breaking through shell redirections and pipes, you may prevent the user from composing the program in the ways they want to compose it. For example,
`yes andersk | python main.py --debug <(echo 'x = raw_input("Name: "); print "Hello,", x') /dev/null`
You also prevent the program from working inside something that isn’t a TTY.

**Pull request closed without merging**